crates/query-engine/sql/src/sql/convert.rs

```diff
@@ -288,15 +277,25 @@ impl Expression {
                 sql.append_syntax(")");
             }
             Expression::FunctionCall { function, args } => {
-                function.to_sql(sql);
-                sql.append_syntax("(");
-                for (index, arg) in args.iter().enumerate() {
-                    arg.to_sql(sql);
-                    if index < (args.len() - 1) {
-                        sql.append_syntax(", ")
+                if function.is_infix {
+                    sql.append_syntax("(");
```

Comment:
I guess there's a nice follow-up PR to add a `parens` helper that adds a parenthesis, runs a continuation, then adds the closer.

Reply:
If we keep BinaryOperation we wouldn't have to worry about this :)

Reply:
Why? Wouldn't this still be a problem for `FunctionCall`?

Reply:
Oh sorry. I think I misunderstood your original suggestion.

I'm usually for combinator style in Haskell, but trying to use the same style in Rust I feel like it's more trouble that it's worth because of Rust's imperative nature and type system. I think this current style works well and that

```rs
sql.append_syntax("(");
<thing>
sql.append_syntax(")");
```

isn't really a problem by any means.